openfga_sdk/client/client.py

```diff
@@ -106,7 +106,7 @@ def set_heading_if_not_set(
         _options["headers"] = {}
 
     if type(_options["headers"]) is dict:
-        if type(_options["headers"].get(name)) not in [int, str]:
+        if _options["headers"].get(name) is None:
```

Comment:
Previously, the `set_heading_if_not_set` method would set the header if it was ANY type except `int` or `str` (including `None`, `dict`, `list`, etc.) This was not intended behavior. It now only sets the header if it's explicitly `None` (not set).

Reply:
Is this warranted? What happens if someone tries to set a map? 🤔 Would we type cast it?